src/sentry/tasks/summaries/daily_summary.py

```diff
@@ -207,12 +207,41 @@ def build_summary_data(
                 group__in=([group for group in regressed_or_escalated_groups]),
                 type__in=(ActivityType.SET_REGRESSION.value, ActivityType.SET_ESCALATING.value),
             )
-            if regressed_or_escalated_groups_today:
-                for activity in regressed_or_escalated_groups_today[:4]:
-                    if activity.type == ActivityType.SET_REGRESSION.value:
-                        project_ctx.regressed_today.append(activity.group)
-                    else:
-                        project_ctx.escalated_today.append(activity.group)
+
+            deduped_groups_by_activity_type: dict[int, set] = {
+                ActivityType.SET_REGRESSION.value: set(),
+                ActivityType.SET_ESCALATING.value: set(),
+            }
+            for activity in regressed_or_escalated_groups_today:
```

Comment:
I think you can simplify this logic a bit:
```py
for activity in regressed_or_escalated_groups_today:
    deduped_groups_by_activity_type[activity.type].setdefault(activity.group, set()).add(activity.group)
```
This would allow you to remove the if else tree on adding / removing the activity group.